internal/pkg/agent/install/uninstall.go

```diff
@@ -46,6 +46,34 @@ func Uninstall(cfgFile, topPath, uninstallToken string, log *logp.Logger, pt *pr
 		return fmt.Errorf("uninstall must be run from outside the installed path '%s'", topPath)
 	}
 
+	ctx := context.Background()
+
+	// check if the agent was installed using --unprivileged by checking the file vault for the agent secret (needed on darwin to correctly load the vault)
+	unprivileged, err := checkForUnprivilegedVault(ctx)
+	if err != nil {
+		return fmt.Errorf("error checking for unprivileged vault: %w", err)
+	}
+
+	// Load config so we can check the tamper protection feature
+	cfg, err := operations.LoadFullAgentConfig(ctx, log, cfgFile, false, unprivileged)
+	if err != nil {
+		return fmt.Errorf("error loading agent config: %w", err)
+	}
+	cfg, err = applyDynamics(ctx, log, cfg)
+	if err != nil {
+		return fmt.Errorf("error applying dynamic inputs: %w", err)
+	}
+	if err := features.Apply(cfg); err != nil {
+		return fmt.Errorf("could not parse and apply feature flags config: %w", err)
+	}
+	// Fail if tamper protection is enabled but no uninstallToken is specified
+	if features.TamperProtection() && uninstallToken == "" {
```

Comment:
This will fail to uninstall when tamper protection is enabled, but endpoint-security is not in the policy, which is stricter than necessary.

This approach does work though. I am curious why when we attempt to uninstall endpoint-security the `install -f` doesn't fail with an empty uninstall token without this change? I am mostly interested in knowing if there is another bug hiding in here that this just skipping over.

I am specifically talking about this call here: https://github.com/elastic/elastic-agent/blob/00d72681762cf96f8fe6ef21bba59e005517d6c5/internal/pkg/agent/install/uninstall.go#L272-L278

That has a sort of similar check, and it should fail with an error if the uninstall token is empty and you try to uninstall endpoint.

https://github.com/elastic/elastic-agent/blob/00d72681762cf96f8fe6ef21bba59e005517d6c5/pkg/component/runtime/service.go#L689-L705

Reply:
I'm not sure. Originally I found that the format of `agent.*` in `elastic-agent inspect` did not match the features flags defined in `features.go` (i've added a top level `agent.protection` attribute), but that was causing it to always expect an unenrollment token

Reply:
> Originally I found that the format of `agent.*` in `elastic-agent inspect` did not match the features flags defined in `features.go`

It would not match, it's two different things: the agent.* is actual setting enabling/disabling the tamper protection for the agent vs the feature flag that disabled/enables the feature code alltogether.

Reply:
> This approach does work though. 

Re-thinking this a bit. This works if you don't specify an uninstall token at all. What does it do if you specify an invalid uninstall token? Does that put us back where we were before this change? This is coming back to the point about this potentially just covering up another bug somewhere further down the stack.